m2cgen/interpreters/interpreter.py

```diff
@@ -71,7 +49,10 @@ class ToCodeInterpreter(BaseToCodeInterpreter):
     about AST.
     """
 
-    with_vectors = False
+    def __init__(self, cg, feature_array_name="input"):
+        super().__init__(cg, feature_array_name=feature_array_name)
+        self.with_vectors = False
+        self._reused_exprs = {}
```

Comment:
nit: Maybe call it `cached_expr_results` or something? `_reused_exprs` gives me a feeling of list or set with somehow actually reused expressions.

Reply:
Fixed